vtex/faststore
# Handle local storage exceeding quota

## What's the purpose of this pull request?

Carrefour reported that their store is crashing for few users. After some investigation, we found out that it's due to the local storage being filled to the maximum, which would crash whenever we would try to add more things into the local storage (with `localStorage.setItem`). This PR bumps the packages that actually solve this problem.

[Read the escalation here.](https://vtex.slack.com/archives/C017NPK8U86/p1620151049393200)

## How it works? 

The `localStorage.setItem` is being wrapped in a `try/catch` statement. On `catch`, it sends an error log to Splunk (so it can help in future investigations) and does nothing else. The only impact that this causes is that we might break offline experience. In order to not prejudice offline experience, we will need to use something other than local storage, but as this bug is **critical** (about 30% fewer orders), we want to move forward with this workaround first.

## How to test it?

https://github.com/vtex-sites/storecomponents.store/pull/822
https://github.com/vtex-sites/marinbrasil.store/pull/477
https://github.com/vtex-sites/btglobal.store/pull/474
https://github.com/vtex-sites/carrefourbrfood.store/pull/442

1. Open the deployment preview of any store and run the following script in the console: `localStorage.buffer = 'a'; for (let i = 0; i < 500000; i++) { localStorage.buffer += localStorage.buffer }`
2. Wait for the script to complete and refresh the page
3. Try to buy anything
4. Assert that the error `Setting the value of '...' exceeded the quota` won't appear on the console

## References

The PRs that really solves the issue:
- https://github.com/vtex/order-manager/pull/14
- https://github.com/vtex/order-items/pull/14
